lib/MusicBrainz/Server/Data/Edit.pm

```diff
@@ -795,6 +796,34 @@ sub _do_accept
 
     my $status = try {
         $edit->accept;
+
+        # If this was not an autoedit and the editor is a beginner
+        # who can now lose the flag, do that
+        unless ($edit->auto_edit) {
+            my $editor_id = $edit->editor_id;
+            my $editor = $self->c->model('Editor')->get_by_id($editor_id);
+            if ($editor->is_beginner && !$editor->is_newbie) {
+                # We check if this will be the tenth accepted edit
+                my $has_nine_edits =
+                    $self->c->sql->select_single_value(<<~"SQL");
+                        SELECT 1
+                          FROM edit
+                         WHERE edit.editor = $editor_id
+                           AND edit.autoedit = 0
+                           AND edit.status = $STATUS_APPLIED
+                        OFFSET 8
+                        SQL
```

Comment:
Sure, I guess - what is the benefit anyway? We're fairly inconsistent with this.

Reply:
Just anti-SQL-injection. These are supposed to be constants, but I'd rather be safe than sorry. :) Interpolating `$STATUS_APPLIED` does seem less bad than `$editor_id`, since it's actually `Readonly`, and `$editor_id` should be validated by Moose as an integer, but...